src/components/privateWrapper/index.js

```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+export const PrivateWrapperComponent = ({ isAuthenticated, children }) => (
+    isAuthenticated && <span>{ children }</ span>
+);
+
+const mapStateToProps = state => ({
+  isAuthenticated: !!state.account.publicKey,
```

Comment:
In private routes, it makes no user-visible difference. But on the login page if user is on slower network, it can longer time for the request and only then user is redirected from login page. These login buttons should not be on login page.